File: src/anylayout/FunctionUtility.java
package anylayout;

import fj.F;
import fj.data.Option;

public final class FunctionUtility
{
    public static <T, R> R constIfNothing(Option<T> o, R or, F<T, R> f)
    {
        return o.option(or, f);
    }

    public static <T> F<T, Integer> minus(final F<T, Integer> f, final F<T, Integer> g)
    {
        return new F<T, Integer>()
        {
            public Integer f(T t)
            {
                return f.f(t)-g.f(t);
            }
        };
    }
}

File: src/anylayout/FunctionUtility.scala
package anylayout;

object FunctionUtility {
 def constIfNothing[T <: AnyRef, R <: AnyRef](o: Option[T], or: R, f: T => R): R = o match { case Some(s) => f(s)
                                                                                             case None => or }

 def minus[T](a: T => Int, b: T => Int): T => Int = t => a(t).intValue - b(t).intValue
}
